Restructure FAQs (#492)
returntocorp/semgrep-docs: docs/faq.md

The [Semgrep Registry](https://semgrep.dev/explore) contains rules from different contributors. Most rules, including all community-contributed rules, are under the [MIT license](https://tldrlegal.com/license/mit-license). Some of the Semgrep Registry rules written by r2c are licensed under the [Semgrep Registry license](../licensing/#semgrep-registry-license).

The source for many r2c-written Registry rules is available at [returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/). Those rules licensed under the Semgrep Registry license cannot be resold without r2c’s permission. Since r2c offers a paid, hosted application (Semgrep App), it’s important to have this restriction so other companies, like major cloud providers, cannot resell r2c rules as a competing service.

### Is it ok to run Semgrep or the r2c rules on my work projects?

Yes! Semgrep is safe to run on your private code. The [Semgrep Registry license’s](../licensing/#semgrep-registry-license) commercial restrictions only come into effect if you are _selling_ the rules provided in the semgrep-rules repository. If that’s the case, you’ll need to talk with r2c first to get permission.

> ### I’m a security professional and want to use the Semgrep rules with my clients as part of my paid services. Is that ok?

Author: Ok, thank you!

> ### I’m a security professional and want to use the Semgrep rules with my clients as part of my paid services. Is that ok?

Author: Changing to:

```md
### I’m a security professional and want to use the [semgrep-rules repository](https://github.com/returntocorp/semgrep-rules) with my clients as part of my paid services. Is that ok?
```

> ### I’m a security professional and want to use the [`semgrep-rules repository`](https://github.com/returntocorp/semgrep-rules) with my clients as part of my paid services. Is that ok?

Reviewer: ```suggestion
### I’m a security professional and want to use the rules from the [`semgrep-rules` repository](https://github.com/returntocorp/semgrep-rules) with my clients as part of my paid services. Is that ok?
```

> If you have a typical consulting service and running Semgrep or Semgrep rules is part of your assessments, that’s great and of course feel free to refer your clients to the [Semgrep App](https://semgrep.dev/). But if your entire service is about scanning code and you want to charge for running the `semgrep-rules` repository that r2c and the community created and maintains, you need to discuss this with r2c.

Reviewer 2: The text is referring to specific public repositories for `semgrep` and `semgrep-rules`. You can also add a link to those repositories here.

Author: Ok! I will change it back and at links to these repositories. @s-santillan

Author: I'm not sure about the word "_running_" in this context @s-santillan. Can we say that somebody is "_running_" `semgrep` repository or `semgrep-rules` repository as a part of their assessments? It sounds a bit strange. What do we actually mean by this (in simpler words)? 

Changing to:

```md
If you have a typical consulting service and running [`semgrep` repository](https://github.com/returntocorp/semgrep) or [`semgrep-rules` repository](https://github.com/returntocorp/semgrep-rules)
```

### Contacting r2c support 

> All users can contact r2c support. Regardless if you a free tier user or higher tier, reach our support through the [r2c Community Slack](https://r2c.dev/slack). Semgrep Team tier customers receive 8\*5 email/phone/Slack support with committed SLAs. See [Support](../support/) for more details.

Reviewer 2: ```suggestion
All users can contact r2c support. Regardless if you are free tier user or higher tier, reach our support through the [r2c Community Slack](https://r2c.dev/slack). Semgrep Team tier customers receive 8\*5 email/phone/Slack support with committed SLAs. See [Support](../support/) for more details.
```

> All users can contact r2c support. Regardless if you are free tier user or higher tier, reach our support through the [r2c Community Slack](https://r2c.dev/slack). Semgrep Team tier customers receive 8\*5 email/phone/Slack support with committed SLAs. See [Support](../support/) for more details.

Reviewer: ```suggestion
All users can contact r2c support. Regardless if you are free tier or paid tier user, reach our support through the [r2c Community Slack](https://r2c.dev/slack). Semgrep Team tier customers receive 8\*5 email/phone/Slack support with committed SLAs. See [Support](../support/) for more details.
```

## Comparisons

### How is Semgrep different from $OTHER\_TOOL or $GENERIC\_[SAST](https://en.wikipedia.org/wiki/Static_application_security_testing)?

Semgrep is an open-source tool with a simple syntax for writing rules: if you can write code, you can write a Semgrep rule — no program analysis Ph.D. required!

To our knowledge, the only other tool with the explicit goal of allowing custom rules is GitHub’s proprietary tool, CodeQL. CodeQL has a domain-specific language that is extremely powerful but is designed for those with significant program analysis expertise, whereas Semgrep is designed for the security engineer or developer who wants to automate code review. Our goal is to make writing a Semgrep rule as easy as copying the code you want to find—and letting the Semgrep engine make the rule and autofix high-quality enough to run in CI or your text editor or IDE.

> Our hosted offering, [Semgrep App](https://semgrep.dev/manage), has a generous free tier (even for private repositories!). It offers a hosted CI integration with a one-click setup so you can start running Semgrep right away. Semgrep's diff-awareness lets you scan new code and doesn’t force you to fix all the existing issues when you first start. For users running inside organizations with many repositories, the hosted offering also offers a policy and notification system that makes it easy to tune Semgrep so that it only reports issues or suggests fixes that get applied. Our goal is a 99% fix rate for what Semgrep reports.

Reviewer 2: I'd say a "one click setup" does not accurately describe the CI process in Semgrep App. Currently it is at 7 clicks (source: June 2022 demos.)

> Our hosted offering, [Semgrep App](https://semgrep.dev/manage), has a generous free tier (even for private repositories!). It offers a hosted CI integration with a one-click setup so you can start running Semgrep right away. Semgrep's diff-awareness lets you scan new code and doesn’t force you to fix all the existing issues when you first start. For users running inside organizations with many repositories, the hosted offering also offers a policy and notification system that makes it easy to tune Semgrep so that it only reports issues or suggests fixes that get applied. Our goal is a 99% fix rate for what Semgrep reports.

Author: ```suggestion
Our hosted offering, [Semgrep App](https://semgrep.dev/manage), has a generous free tier (even for private repositories!). It offers a hosted CI integration with a quick setup so you can start running Semgrep right away. Semgrep's diff-awareness lets you scan new code and doesn’t force you to fix all the existing issues when you first start. For users running inside organizations with many repositories, the hosted offering also offers a policy and notification system that makes it easy to tune Semgrep so that it only reports issues or suggests fixes that get applied. Our goal is a 99% fix rate for what Semgrep reports.
```

And many more: see [the Semgrep philosophy](../contributing/semgrep-philosophy/) for further reading.

> ### Comparing Semgrep to other linters

Reviewer: ```suggestion
### Comparing Semgrep to linters
```
